File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode createBinaryTree(int[][] descriptions) {
        Map<Integer,TreeNode>mpp=new HashMap<>();
        Set<Integer> children=new HashSet<>();
        for(int node[]:descriptions){
            int parent=node[0];
            int value=node[1];
            int isleft=node[2];
            children.add(value);

            if(mpp.containsKey(parent)){
                if(mpp.containsKey(value)){
                    if(isleft==1) mpp.get(parent).left=mpp.get(value);
                    else mpp.get(parent).right=mpp.get(value);
                }else{
                    TreeNode temp=new TreeNode(value);
                    if(isleft==1) mpp.get(parent).left=temp;
                    else mpp.get(parent).right=temp;
                    mpp.put(value,temp);
                }
            }else{
                TreeNode parentNode=new TreeNode(parent);
                mpp.put(parent,parentNode);
                if(mpp.containsKey(value)){
                    if(isleft==1) mpp.get(parent).left=mpp.get(value);
                    else mpp.get(parent).right=mpp.get(value);
                }else{
                    TreeNode temp=new TreeNode(value);
                    if(isleft==1) mpp.get(parent).left=temp;
                    else mpp.get(parent).right=temp;
                    mpp.put(value,temp);
                }
            }
        }

        for(int it:mpp.keySet()){
            if(!children.contains(it)) return mpp.get(it);
        }
        return null;
    }
}

File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* createBinaryTree(vector<vector<int>>& descriptions) {
        unordered_map<int,TreeNode*>mpp;
        unordered_set<int> children;
        for(vector<int>node:descriptions){
            int parent=node[0];
            int value=node[1];
            int isleft=node[2];
            children.insert(value);

            if(mpp.count(parent)){
                if(mpp.count(value)){
                    if(isleft) mpp[parent]->left=mpp[value];
                    else mpp[parent]->right=mpp[value];
                }else{
                    TreeNode* temp=new TreeNode(value);
                    if(isleft) mpp[parent]->left=temp;
                    else mpp[parent]->right=temp;
                    mpp[value]=temp;
                }
            }else{
                TreeNode* parentNode=new TreeNode(parent);
                mpp[parent]=parentNode;
                if(mpp.count(value)){
                    if(isleft) mpp[parent]->left=mpp[value];
                    else mpp[parent]->right=mpp[value];
                }else{
                    TreeNode* temp=new TreeNode(value);
                    if(isleft) mpp[parent]->left=temp;
                    else mpp[parent]->right=temp;
                    mpp[value]=temp;
                }
            }
        }

        for(auto it:mpp){
            if(children.count(it.first)==0) return it.second;
        }
        return nullptr;
    }
};